### PR TITLE
Initial commit for the webhook to set the annotations about mutator.

### PR DESCRIPTION
**Pull request by vagababov**
The user that created or updated the resource will be set in the
annotations.

Part of knative/serving#1642.

/cc @mattmoor

### CODE REVIEW

File: apis/interfaces.go [hunk 1]
```diff
@@ -47,3 +47,8 @@ type Listable interface {
 
 	GetListType() runtime.Object
 }
+
+// Annotatable indicates that a particular type applies various annotations.
+type Annotatable interface {
+	AnnotateUserInfo(previous Annotatable, userName string)
```
Comment (mattmoor) [inline apis/interfaces.go]: I'd pass in the [UserInfo](https://godoc.org/k8s.io/api/authentication/v1#UserInfo)

File: webhook/webhook.go [hunk 2]
```diff
@@ -126,6 +127,7 @@ type AdmissionController struct {
 type GenericCRD interface {
 	apis.Defaultable
 	apis.Validatable
+	apis.Annotatable
```
Comment (mattmoor) [inline webhook/webhook.go]: I wouldn't require this, I'd check this like we do for `apis.Immutable`

Comment (vagababov) [reply webhook/webhook.go]: Done.

File: apis/interfaces.go (hunk 1 above)
Comment (vagababov) [reply apis/interfaces.go]: Hm, from one point of view sure, from another, do we have ideas how to utilize other fields?

Comment (mattmoor) [reply apis/interfaces.go]: What if I just want the UID for PII reasons :)

**Pull request merged**